src/Migrations/M032MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,203 @@
+                }
+
+                $block_attrs = array_merge(['columns_block_style' => 'image'], $block_attrs);
+                $block_attrs['className'] = 'is-style-image';
+                $block = self::transform_block($block_attrs);
+            }
+
+            // Unset the reference to prevent potential issues.
+            unset($block);
+
+            // Serialize the blocks content.
+            $new_content = serialize_blocks($blocks);
+
+            if ($post->post_content !== $new_content) {
+                $post_update = array(
+                    'ID' => $post->ID,
+                    'post_content' => $new_content,
+                );
+
+                try {
+                    // Update the post with the replaced blocks.
+                    wp_update_post($post_update);
+                    $result = true;
+                } catch (\Throwable $e) {
+                    echo 'Error on post ', $post->ID, "\n";
+                    echo $e->getMessage(), "\n";
+                }
+            }
+
+            echo $result
+                ? "Migration successful\n"
+                : "Migration wasn't executed\n"; // phpcs:ignore
+        }
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Get all the posts using block name.
+     *
+     * @param string $block_name - A block name.
+     * @return mixed - The posts using block or null if no posts are found.
+     */
+    private static function get_posts_using_block(string $block_name): mixed
+    {
+        $search = new BlockSearch();
+
+        $post_ids = $search->get_posts_with_block($block_name);
+
+        if (empty($post_ids)) {
+            return null;
+        }
+
+        $args = [
+            'include' => $post_ids,
+            'post_type' => self::POST_TYPES,
+        ];
+
+        $posts = get_posts($args) ?? [];
+
+        if (empty($posts)) {
+            return null;
+        }
+
+        return $posts;
+    }
+
+    /**
+     * Transform a block attrs into columns block.
+     *
+     * @param array $block_attrs - A block attrs array.
+     * @return array - The transformed block.
+     */
+    private static function transform_block(array $block_attrs): array
+    {
+        $p4_columns_block['blockName'] = 'planet4-blocks/columns';
+        $p4_columns_block['attrs'] = $block_attrs;
+        $p4_columns_block['innerBlocks'] = [];
+        $p4_columns_block['innerHTML'] = '';
+        $p4_columns_block['innerContent'] = [];
+
+        return $p4_columns_block;
+    }
+
+    /**
+     * Get the p4 columns block attrs.
+     *
+     * @param array $block - A parsed cover block.
+     * @return array - A P4 columns block attrs.
+     */
+    private static function get_columns_block_attrs(array $block): array
+    {
+        $block_attrs = [];
+        foreach ($block['attrs'] as $key => $value) {
+            if (empty($value)) {
+                continue;
+            }
+
+            if ('title' === $key) {
```

Comment:
Do we need the `if` and `else if` statements here?
Since we are only looking for the title, description, and tags maybe we could do something like this to improve the readability:

```php
private static function get_columns_block_attrs(array $block): array
    {
        $block_attrs = [];
        $block_attrs['columns_title'] = $block['attrs']['title'] ?? '';
        $block_attrs['columns_description'] = $block['attrs']['description'] ? : str_replace("\n", "\u003cbr\u003e", $block['attrs']['description']); '';
         ...
```